src/2020-06-08-mixing-modules-and-export.Rmd

```diff
@@ -0,0 +1,106 @@
+---
+title: "Mixing Rcpp modules and Rcpp attributes"
+author: "Bob Jansen"
+license: GPL (>= 2)
+tags: basics attributes modules
+summary: How to use Rcpp modules in combination with Rcpp attributes
+---
+
+### Introduction
+
+With Rcpp modules (Eddelbuettel and Francois, 2017) it is easy to expose
+\proglang{C++} classes and functions to \proglang{R}. This note describes how
+to use exported classes in conjunction with functions exported using Rcpp
+attributes (Eddelbuettel and Francois, 2017) using Rcpp attributes
+(Eddelbuettel and Francois, 2017) through the use of
```

Comment:
It's nit picky but if you look in the Rcpp package for Rcpp.bib you see the proper (as of last update to it) references the vignettes.  Attributes is IIRC "Allaire et al".

Reply:
Typed these in, it's not customary to provide a bibliography at the end right?

Reply:
Good question.  We could, but given that these pieces are a) generally very short and b) in Markdown (!!) so we could just link it feels overly formal. 

I am also a diehard pdf user for formal writeups so I don't really know what best practices is with pandoc and pandoc-cite if one does _not_ go to pdf mode...  

Any suggestions?